packages/catalog-process/src/services/catalogService.ts

```diff
@@ -890,14 +894,52 @@ export function catalogServiceBuilder(
         descriptors: [...eservice.data.descriptors, newDescriptor],
       };
 
-      const event = toCreateEventEServiceDescriptorAdded(
+      const descriptorCreationEvent = toCreateEventEServiceDescriptorAdded(
         newEservice,
-        eservice.metadata.version,
+        eserviceVersion,
         descriptorId,
         correlationId
       );
-      await repository.createEvent(event);
 
+      // eslint-disable-next-line functional/no-let
+      let events: Array<CreateEvent<EServiceEvent>> = [descriptorCreationEvent];
+
+      for (const document of eserviceDescriptorSeed.docs) {
```

Comment:
minor, but if we use a reduce on `eserviceDescriptorSeed.docs` we should be able to avoid mutability on `eserviceVersion`, `newDescriptor` and `events`

Reply:
Updated